docs/peg_oled_display.md

```diff
@@ -0,0 +1,97 @@
+# Peg Oled Display
+To use this you need to make some changes to your kb.py as well as you main.py I will break it into two sections for you. 
+
+### What you can and cant do
+
+#### Can do
+* display images
+* display text
+* set images or text to react to your layer
+
+
+#### Cant do yet / on the way
+* react to battery percentage
+* react to WPM 
+
+## Required Libs
+you need these frozen into your circuitpython or in a lib folder at the root of your drive.
+* [Adafruit_CircuitPython_DisplayIO_SSD1306](https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_SSD1306)
+* [Adafruit_CircuitPython_Display_Text](https://github.com/adafruit/Adafruit_CircuitPython_Display_Text)
+* [download .mpy versions from here](https://github.com/adafruit/Adafruit_CircuitPython_Bundle/releases/download/20220415/adafruit-circuitpython-bundle-7.x-mpy-20220415.zip)
+
+
+## kb.py
+Your chosen board may already have these changes done If not you will need to add them.
+
+you need to add SCL and SDA to your keyboard. The pins the oled are connected to on your controller may not be called "SCL"and "SDA" they could be "GP21" and "GP13" for example. The best way to find out what they are is look at the boards pinout.
+```python
+    SCL=board.SCL
+    SDA=board.SDA
+```
+
+
+## Main.py
+These are the changes that need to be made / added to your main.py
+### Config
+no mater how you are going to use the oled you need this part
+```python
+    from kmk.extensions.peg_oledDisplay import oled
+    keyboard = KMKKeyboard()
+    # ... Other oled code
+    keyboard.extensions.append(oled_ext) 
+
+```
+### Photos
+So the config for photos is quite simple. Getting the photos maybe not so much. I will explain.
+
+oled takes 3-4 arguments 
+1. keyboard
+    * this is where we get the config from
+2.  array of data

```

Comment:
I'm making these comments in reverse order, I'm now realizing, but fine, whatever: if the data is this structured, it probably shouldn't be an anonymous array. it should be a class of its own outright. (in desktop Python I'd also suggest a `namedtuple` but we no longer use those here because their implementation [sometimes takes an external dependency](https://github.com/KMKfw/kmk_firmware/pull/203#discussion_r654896112))